omegaconf/_utils.py

```diff
@@ -454,6 +439,16 @@ def is_primitive_container(obj: Any) -> bool:
     return is_primitive_list(obj) or is_primitive_dict(obj)
 
 
+def get_union_types(ref_type: Optional[Any]) -> List[Any]:
+    args = getattr(ref_type, "__args__", None)
+    element_types: List[Any]
+    if ref_type is not Union and args is not None and args[0] is not Any:
```

Comment:
I've changed it to return Any in that case, tell me what you think.

Reply:
> I've changed it to return Any in that case, tell me what you think.

I think an error would be better, otherwise you have to deal with the return value of `get_union_types` being `Any` everywhere else, and this would make the code more complex for no added value. I see no reason to support it (it's more likely to hide problems than to enable more functionalities).
(also, just trying the latest code (720e405) I'm running into a crash when using `x: Union[Any, int] = 1`)

Reply:
I added an exception for Any as element type. The same goes to `Union[None, vt]` and `Union[vt]`